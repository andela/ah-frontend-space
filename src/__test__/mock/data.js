export const data = {
  user: {
    username: '',
    email: '',
    auth_token: '',
  },
};

export const user = {
  email: 'test@email.com',
  password: '12345Enter',
};

export const errors = {
  errors: {
    error: 'An error occured',
  },
};

export const state = {};

export const registeredUser = {
  data: {
    email: 'myemail@email.com',
    password: '123456',
  },
};

export const mockArticle = {
  title: 'Lorem ipsum dolor sit amet',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus ac ipsum tortor. Quisque at rhoncus ipsum. Vivamus pulvinar ultrices odio vitae faucibus. Mauris congue sit amet orci a gravida. Integer elementum elit vel massa dapibus lobortis. Phasellus laoreet, metus fermentum sagittis faucibus, tortor urna dignissim augue, et mattis felis magna euismod urna. Aenean lorem arcu, porta id auctor id, consequat at lorem.',
  body: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus ac ipsum tortor. Quisque at rhoncus ipsum. Vivamus pulvinar ultrices odio vitae faucibus. Mauris congue sit amet orci a gravida. Integer elementum elit vel massa dapibus lobortis. Phasellus laoreet, metus fermentum sagittis faucibus, tortor urna dignissim augue, et mattis felis magna euismod urna. Aenean lorem arcu, porta id auctor id, consequat at lorem.
  Nulla facilisi. Morbi hendrerit sollicitudin neque, eget scelerisque purus rutrum vitae. Ut id rutrum ex, quis semper justo. Nullam sed libero in nulla ultricies sagittis. Fusce et lacus sem. Nam egestas sem dictum sem pretium pulvinar. Maecenas convallis neque ex, nec ultrices elit interdum sit amet. Curabitur ut fermentum erat. Donec mauris augue, mattis egestas rutrum ut, condimentum quis massa. Suspendisse potenti. Suspendisse aliquet dolor justo, eu euismod orci fermentum ac. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Quisque vitae elementum lorem, nec euismod purus. Curabitur tellus est, fringilla sed arcu in, pulvinar rhoncus metus. Sed aliquam risus mi, a placerat justo efficitur eu. Donec varius velit condimentum viverra placerat.
  Praesent ut quam vitae libero faucibus feugiat. Aenean hendrerit pulvinar laoreet. Morbi egestas interdum augue, sed tempor metus malesuada vel. Fusce et leo est. Nam imperdiet velit a nibh semper, nec imperdiet magna convallis. Quisque nulla dui, pellentesque eget orci quis, euismod placerat neque. Sed lacinia, lectus quis finibus vestibulum, velit mauris pretium magna, ut gravida velit ex eu ipsum. Sed nec eros vel neque mattis commodo nec non diam. Quisque faucibus ipsum eget neque ultricies luctus. Donec eget risus eu ante hendrerit auctor sit amet at lorem. Integer quis egestas risus. Morbi et sem nunc. Donec nec diam in dui posuere interdum ac at metus. Ut vestibulum eleifend tellus, non ornare sapien posuere a.
  Vivamus tristique neque in dui ornare, id egestas elit consectetur. Aenean et metus imperdiet, lobortis ligula ac, vestibulum felis. Curabitur eleifend, purus quis cursus tempor, lacus purus pellentesque nisl, ac volutpat nibh lorem id libero. Morbi id ante enim. Aenean tincidunt lacus urna, eget vulputate neque iaculis id. Proin vestibulum mi eget pellentesque varius. Nunc ante sapien, lobortis ac massa nec, fermentum accumsan ipsum. Suspendisse eu semper orci. Phasellus finibus eu leo sit amet tempus. Suspendisse dictum nisl eget nisl interdum, a pulvinar sapien euismod. Mauris augue odio, ultricies id tincidunt id, hendrerit ac lacus. Vivamus ac nulla vel dolor tincidunt interdum eget in tellus. In congue venenatis mi et accumsan. Aenean luctus ullamcorper dolor, ac condimentum nibh feugiat nec.
  Nulla elit nunc, sollicitudin et leo tempus, convallis mattis magna. Sed eget magna in enim semper blandit ut at est. Vivamus fermentum varius arcu, non facilisis lorem mollis nec. Ut sodales suscipit enim, vel auctor sapien volutpat et. Mauris ac ex lacus. Donec finibus eget eros in laoreet. Etiam nec imperdiet justo, id porttitor justo. Donec vitae ullamcorper est.`,
};
