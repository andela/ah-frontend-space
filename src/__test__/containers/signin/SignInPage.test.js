/* eslint-disable no-shadow */
/* eslint-disable prefer-promise-reject-errors */
import React from 'react';
import { shallow } from 'enzyme';
import {
  SignInPage,
  mapStateToProps,
  mapDispatchToProps,
} from '../../../containers/signIn/SignInPage';
import { user, errors } from '../../mock/data';

describe('render SignInPage with shallow', () => {
  const defaultProps = {
    user: { status: 200 },
    errors: {},
    actions: {
      userSignIn: user => new Promise((resolve, reject) => {
        if (user.status === 200) {
          resolve('Success Message');
        } else {
          reject('Not Successful');
        }
      }),
    },
    loading: false,
  };

  const wrapper = shallow(<SignInPage {...defaultProps} />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should map state to props', () => {
    const state = {
      user: {
        email: 'email@email.com',
        password: 'enter123',
      },
      errors: {},
      loading: false,

    };
    expect(mapStateToProps(state)).toEqual({
      user: {
        email: 'email@email.com',
        password: 'enter123',
      },
      errors: {},
      loading: false,
    });
  });

  it('should map dispatch to the signInSuccess action and return a data object ', () => {
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).actions.signInSuccess(user);
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'SIGNIN_SUCCESS', data: user });
  });

  it('should map dispatch to the signInFailure action and return a errors object ', () => {
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).actions.signInFailure(errors);
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'SIGNIN_FAILURE', error: errors });
  });

  it('should return true if form is invalid ', () => {
    const wrapperInstance = wrapper.instance();
    expect(wrapperInstance.formIsInValid()).toEqual(true);
  });

  it('should return false if form is valid ', () => {
    wrapper.setState({
      user: {
        email: 'test@email.com',
        password: '1234Enter',
        status: 200,
      },
      errors: { onSave: false },
    });
    const wrapperInstance = wrapper.instance();
    expect(wrapperInstance.state.user.email).toBe('test@email.com');
    expect(wrapperInstance.formIsInValid()).toEqual(false);
  });

  it('should handle submit successfully', () => {
    wrapper.setState({
      user: {
        email: 'test@email.com',
        password: '1234Enter',
        status: 200,
      },
      errors: { onSave: false },
    });
    const wrapperInstance = wrapper.instance();
    const event = { preventDefault: jest.fn() };
    jest.spyOn(wrapperInstance, 'onSubmitHandler');
    wrapperInstance.onSubmitHandler(event);

    expect(wrapperInstance.onSubmitHandler).toBeCalled();
  });


  it('should handle onChange event', () => {
    wrapper.setState({
      target: {
        name: 'password',
        value: '1234Enter',
      },
      errors: { onSave: false },
    });
    const wrapperInstance = wrapper.instance();
    const event = {
      target: {
        name: 'password',
        value: '1234Enter',
      },
    };
    jest.spyOn(wrapperInstance, 'onChangeHandler');
    wrapperInstance.onChangeHandler(event);

    expect(wrapperInstance.onChangeHandler).toBeCalled();
  });
});
