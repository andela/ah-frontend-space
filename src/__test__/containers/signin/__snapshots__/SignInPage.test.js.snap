// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SignInPage with shallow should match snapshot 1`] = `ShallowWrapper {}`;
